src/realm/db.cpp

```diff
@@ -849,6 +848,16 @@ void DB::do_open(const std::string& path, bool no_create_file, bool is_backend,
 #else
         m_file.lock_shared(); // Throws
 #endif
+        // The coordination/management dir is created as a side effect of the lock
+        // operation above if needed for lock emulation. But it may also be needed
+        // for other purposes, so make sure it exists:
+        if (!util::File::is_dir(m_coordination_dir)) {
```

Comment:
This check is at most just a pessimization. Checking if the directory exists isn't any faster than attempting to create it in the case where it does exist, and checking then creating is slower if it doesn't. There should be no difference in correctness.

Reply:
Right on. Will fix.